diff --git a/fs/unicode/utf8-norm.c b/fs/unicode/utf8-norm.c
index 768f8ab..fadb72b 100644
--- a/fs/unicode/utf8-norm.c
+++ b/fs/unicode/utf8-norm.c
@@ -513,21 +513,18 @@ int utf8byte(struct utf8cursor *u8c)
 		    u8c->um->ntab[u8c->n]->maxage) {
 			ccc = STOPPER;
 		} else if (ccc == DECOMPOSE) {
-			u8c->len -= utf8clen(u8c->s);
-			u8c->p = u8c->s + utf8clen(u8c->s);
-			u8c->s = LEAF_STR(leaf);
-			/* Empty decomposition implies CCC 0. */
-			if (*u8c->s == '\0') {
-				if (u8c->ccc == STOPPER)
-					continue;
-				ccc = STOPPER;
-				goto ccc_mismatch;
-			}
-
-			leaf = utf8lookup(u8c->um, u8c->n, u8c->hangul, u8c->s);
-			if (!leaf)
-				return -1;
-			ccc = LEAF_CCC(leaf);
+			if (*LEAF_STR(leaf) == '\0') {
+                            ccc = STOPPER; // 只设STOPPER，不推进
+                        } else {
+                        // 非空字符，才执行指针移动和状态修改
+                        u8c->len -= utf8clen(u8c->s);
+                        u8c->p = u8c->s + utf8clen(u8c->s);
+                        u8c->s = LEAF_STR(leaf);
+                        leaf = utf8lookup(u8c->um, u8c->n, u8c->hangul, u8c->s);
+                        if (!leaf)
+                            return -1;
+                        ccc = LEAF_CCC(leaf);
+                    }
 		}
 
 		/*
